e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,175 @@
+
+                var endpoint = ((string)json.items[indexOfTestPlc].discoveryUrls[0]).TrimEnd('/');
+                if (TestHelper.IsUrlStringsEqual(endpoint, testPlc.EndpointUrl)) {
+                    found = true;
+                    break;
+                }
+            }
+
+            Assert.True(found, "OPC Application not activated");
+
+            if (string.IsNullOrWhiteSpace(_context.OpcUaEndpointId)) {
+                WaitForEndpointDiscoveryAndStoreInContext();
+                Assert.False(string.IsNullOrWhiteSpace(_context.OpcUaEndpointId), "The endpoint was not set");
+            }
+
+            TestHelper.Registry_ActivateEndpointAsync(_context, cts.Token).GetAwaiter().GetResult();
+        }
+
+        [Fact, PriorityOrder(102)]
+        public void D1_Discover_OPC_UA_Endpoints() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var endpoints = TestHelper.Registry_GetEndpointsAsync(_context, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.SingleOrDefault(e => string.Equals(_context.OpcUaEndpointId, e.Id));
+
+            Assert.NotNull(endpoint.Id);
+            Assert.Equal(TestConstants.StateConstants.ActivatedAndConnected, endpoint.ActivationState);
+            Assert.Equal(TestConstants.StateConstants.Ready, endpoint.EndpointState);
+        }
+
+        [Fact, PriorityOrder(103)]
+        public void T1_Browse_Address_Space() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+        }
+
+        [Fact, PriorityOrder(104)]
+        public void T1_1_BrowseSpecificNode() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            var nodeId = nodes.First().NodeId;
+
+            nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, nodeId).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+        }
+
+        [Fact, PriorityOrder(105)]
+        public void T1_2_BrowseAllNodesOfTypeObject() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpoint_RecursiveAsync(_context, _context.OpcUaEndpointId, "Object", null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            Assert.True(nodes.Count > 150);
```

Comment:
Should we validate that the retrieved nodes indeed have the type "Object"?

Reply:
This filtering is currently done in TestHelper, so that would just validate the test code.